CONTRIBUTING.rst

```diff
@@ -0,0 +1,313 @@
+* Describe the exact behavior you would expect the suggested feature to produce.
+* Provide sample code that you would use to access the feature. If possible, include code for how you think the feature could be built into pyiron's codebase. Include code snippets as markdown code blocks.
+
+Your first code contribution
+----------------------------
+
+Unsure where to begin contributing to pyiron? You can start by looking
+through these good-first-issue and help-wanted issues:
+
+* `Good first issues`_ - issues which should only require a few lines of code, and a test or two.
+* `Help wanted issues`_ - issues which should be a bit more involved than beginner issues.
+
+**Local development**
+
+pyiron can be developed and tested locally. If you are using pyiron to run an
+external software package, e.g. VASP or LAMMPS, you might also need to install
+those packages locally to run certain integration tests in pyiron.
+
+To get the developmental (git) version of pyiron,
+
+```
+git clone https://github.com/pyiron/pyiron.git
+```
+If you don't already have the dependencies installed in your environment,
+```
+cd pyiron
+pip install .
+```
+
+Pull requests
+-------------
+
+The process described here has several goals:
+
+* Maintain pyiron's quality
+* Fix problems that are important to users
+* Engage the community in working toward the best possible tools
+* Enable a sustainable system for pyiron's maintainers to review contributions
+
+Please follow these steps to have your contribution considered by the maintainers:
+
+* Keep the changes in your pull request as focused as possible- only address one issue per pull request wherever possible.
+* Follow the `Styleguides`_
+* Assign the appropriate label (see `Issue and pull request labels`_) to your pull request. If you are fixing a specific Github issue, reference the issue directly in the pull request comments.
+* If you are aware which maintainer is most closely related to the code you've edited, feel free to request their review.
+* After you submit your pull request, verify that all status checks are passing.
+* If a status check fails and it seems to be unrelated to your changes, explain why the failure is unrelated as a comment in your pull request.
+
+While the prerequisites above must be satisfied prior to having your
+pull request reviewed, the reviewer(s) may ask you to complete
+additional design work, tests, or other changes before your pull
+request can be ultimately accepted.
+
+Styleguides
+===========
+
+Git commit messages
+-------------------
+
+* Use the present tense ("Add feature" not "Added feature")
+* Use the imperative mood ("Move cursor to..." not "Moves cursor to...")
+* Limit the first line to 72 characters or less
+* Reference issues and pull requests liberally after the first line
+* When only changing documentation, include [ci skip] in the commit title
+* Consider starting the commit message with an applicable emoji:
```

Comment:
For pycharm users, you can install the [emoji plugin](https://plugins.jetbrains.com/plugin/9174-emoji-support-plugin) to add emojis for your commit messages: 👨‍🎨 🐎 🍰